crates/pgt_workspace/src/workspace/server/parser.rs

```diff
@@ -0,0 +1,374 @@
+use std::sync::Arc;
+
+use pgt_diagnostics::serde::Diagnostic as SDiagnostic;
+use pgt_fs::PgTPath;
+use pgt_query_ext::diagnostics::SyntaxDiagnostic;
+use pgt_text_size::{TextRange, TextSize};
+
+use crate::workspace::ChangeFileParams;
+
+use super::{
+    change::StatementChange,
+    document::{Document, StatementIterator},
+    pg_query::PgQueryStore,
+    sql_function::SQLFunctionBodyStore,
+    statement_identifier::StatementId,
+    tree_sitter::TreeSitterStore,
+};
+
+pub struct Parser {
```

Comment:
`Parser` is a bit of an odd name to me, since it isn't used to do parsing, but as more of a container for the results.

Is this maybe a `ParsedDocument` ? Or a `ParsedStatements` struct?